packages/block-editor/src/components/block-list/block.js

```diff
@@ -505,9 +505,14 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, registry ) => {
 			) {
 				__unstableMarkLastChangeAsPersistent();
 			}
+			//Unsynced patterns are nested in an array so we need to flatten them.
```

Comment:
I was initially flattening this [in the autocomplete component](https://github.com/WordPress/gutenberg/pull/52866/commits/49a9e920a9a6bf71ff8e2240b09144fd6389f501#diff-2293292098a269e3c8c0facd2a6472b365dc5b7814ef7a3047e9e09f579f56a3R110), but decided there were more potential backwards compat issues modifying that public API. 

It seems like this `replaceBlocks` API is currently only ever expecting an array with block objects rather than an array with a nested array of blocks, so this seems like a safe extension, but open to thoughts about whether this is the best approach.